main.tf

```diff
@@ -25,4 +25,7 @@ resource "google_dataflow_job" "dataflow_job" {
   temp_gcs_location     = "gs://${var.temp_gcs_location}/tmp_dir"
   parameters            = "${var.parameters}"
   service_account_email = "${var.service_account_email}"
+  network               = "${var.network_name}"
+  subnetwork            = "${replace(var.subnetwork_self_link, "/(.*)/regions/(.*)/", "regions/$2")}"
```

Comment:
Is the `self_link` incompatible with the expected format?

Reply:
Yeah the format is: "regions/REGION/subnetworks/SUBNETWORK" which is only  part of the self_link.
See [here](https://www.terraform.io/docs/providers/google/r/dataflow_job.html#subnetwork)